bindings/cpu_profiler.cc

```diff
@@ -132,45 +208,58 @@ class Profiler {
   static void DeleteInstance(void* data);
 };
 
-void Profiler::DeleteInstance(void* data) {
-  Profiler* profiler = static_cast<Profiler*>(data);
-  profiler->cpu_profiler->Dispose();
-  delete profiler;
-}
-
 class SentryProfile {
 private:
-  uint64_t started_at = 0;
+  uint64_t started_at;
   uint16_t heap_measurement_index = 0;
-  std::vector<uint64_t> heap_stats_ts;
-  std::vector<uint64_t> heap_stats_usage;
-  std::function<void(uv_timer_t*, uint64_t, v8::HeapStatistics&)> memory_sampler_cb;
-  const char* id;
-  ProfileStatus status;
+  uint16_t cpu_measurement_index = 0;
+
+  std::vector<uint64_t> heap_stats_ts = std::vector<uint64_t>(300);
+  std::vector<uint64_t> heap_stats_usage = std::vector<uint64_t>(300);
+
+  std::vector<uint64_t> cpu_stats_ts = std::vector<uint64_t>(300);
+  std::vector<double> cpu_stats_usage = std::vector<double>(300);
+
+  std::function<void(uint64_t, v8::HeapStatistics&)> memory_sampler_cb;
+  std::function<void(uint64_t, double)> cpu_sampler_cb;
+
+  ProfileStatus status = ProfileStatus::kNotStarted;
+  std::string id;
 
 public:
   explicit SentryProfile(const char* id) :
     started_at(uv_hrtime()),
-    heap_measurement_index(0),
-    heap_stats_ts(300),
-    heap_stats_usage(300),
-    memory_sampler_cb([this](uv_timer_t* timer, uint64_t ts, v8::HeapStatistics& stats) {
-    heap_stats_ts.insert(heap_stats_ts.begin() + heap_measurement_index, ts - started_at);
-    heap_stats_usage.insert(heap_stats_usage.begin() + heap_measurement_index, static_cast<uint64_t>(stats.used_heap_size()));
-    ++heap_measurement_index;
-      }), id(id), status(ProfileStatus::kNotStarted) {}
-
-
-      const std::vector<uint64_t>& heap_usage_timestamps();
-      const std::vector<uint64_t>& heap_usage_values();
-      const uint16_t& heap_usage_entries_count();
-
-      void Start(Profiler* profiler);
-      v8::CpuProfile* Stop(Profiler* profiler);
+
+    memory_sampler_cb([this](uint64_t ts, v8::HeapStatistics& stats) {
+      heap_stats_ts.insert(heap_stats_ts.begin() + heap_measurement_index, ts - started_at);
+      heap_stats_usage.insert(heap_stats_usage.begin() + heap_measurement_index, static_cast<uint64_t>(stats.used_heap_size()));
+      ++heap_measurement_index;
+    }),
+
+    cpu_sampler_cb([this](uint64_t ts, double rate) {
+      cpu_stats_ts.insert(cpu_stats_ts.begin() + cpu_measurement_index, ts - started_at);
+      cpu_stats_usage.insert(cpu_stats_usage.begin() + cpu_measurement_index, rate);
+      ++cpu_measurement_index;
+    }),
+      
+    status(ProfileStatus::kNotStarted),
+    id(id) {}
+
+  const std::vector<uint64_t>& heap_usage_timestamps();
+  const std::vector<uint64_t>& heap_usage_values();
+  const uint16_t& heap_usage_entries_count();
+
+  const std::vector<uint64_t>& cpu_usage_timestamps();
+  const std::vector<double>& cpu_usage_values();
+  const uint16_t& cpu_usage_entries_count();
```

Comment:
same comment as in the previous PR, there should be another `const` at the **end** of the function signature to indicate that the function does not mutate - more details here: https://learn.microsoft.com/en-us/cpp/cpp/const-cpp?view=msvc-170#const-member-functions